lib/Solver/Z3Builder.cpp

```diff
@@ -67,8 +73,17 @@ void Z3ArrayExprHash::clear() {
   _array_hash.clear();
 }
 
-Z3Builder::Z3Builder(bool autoClearConstructCache)
-    : autoClearConstructCache(autoClearConstructCache) {
+Z3Builder::Z3Builder(bool autoClearConstructCache, const char* z3LogInteractionFileArg)
+    : autoClearConstructCache(autoClearConstructCache), z3LogInteractionFile("") {
+  if (z3LogInteractionFileArg)
+    this->z3LogInteractionFile = std::string(z3LogInteractionFileArg);
+  if (z3LogInteractionFile.length() > 0) {
+    llvm::errs() << "Logging Z3 API interaction to \"" << z3LogInteractionFile
```

Comment:
The Z3Solver is also used in the `kleaver` tool. Are you sure you want this change? Same applies for the other comments.